aprsis/aprsis.go

```diff
@@ -68,18 +70,21 @@ func (a *APRSIS) SetInfoHandler(to InfoHandler) {
 	a.infoHandler = to
 }
 
-// Dial an APRS-IS service.
-func Dial(prot, addr string) (rv *APRSIS, err error) {
-	var conn *textproto.Conn
-	conn, err = textproto.Dial(prot, addr)
-	if err != nil {
-		return
+// ManageConnection - Goroutine that receives APRS stream and sends it to bound channel
+func (a *APRSIS) ManageConnection() {
+	for {
+		frame, err := a.Next()
+		if err != nil {
+			log.Println(err)
+			break
```

Comment:
`return err` would be clearer here -- this API sort of requires log-and-continue.  Returning the error would be cleaner, but you'd have to arrange for the management differently.  However, that's what `a.Next()` intended.

We tend to avoid exposing channels in APIs for this reason -- it gets very complicated to cleanly manage all of the details.  It's easy enough for a particular application to convert something like `a.Next()` into something that feeds data down a channel (as you've done here), but it's hard to properly handle all the parts the other way around.

Reply:
I'm still thinking this through...Some kind of backoff/retry would be nice. Other than that, I think, this PR can be merged.